### PR TITLE
Continuous upgrade job for CICD cluster

### PR DESCRIPTION
**Pull request by jhadvig**
@stevekuznetsov PTAL

### CODE REVIEW

File: continuous-upgrade/jobs/cicd-upgrade-ocp.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+                  --become-user root     \
+                  --connection local     \
+                  --inventory sjb/inventory/ \
+                   "/usr/share/ansible/openshift-ansible/playbooks/byo/openshift-cluster/upgrades/v3_${ATOMIC_OPENSHIFT_UPGRADE_RELEASE_MINOR_VERSION=6}/upgrade.yml"     \
+                  -e etcd_data_dir="/tmp/etcd" \
+                  -e openshift_pkg_version="-\${ATOMIC_OPENSHIFT_UPGRADE_RELEASE_VERSION}" 
+
+SCRIPT
+chmod +x "${script}"
+scp -F ~jenkins/.ssh/cicd.cluster_key  "${script}" root@master1.cicd.openshift.com:"${script}"
```
Comment (stevekuznetsov) [inline continuous-upgrade/jobs/cicd-upgrade-ocp.sh]: `-F` is for `.ssh_config` files, you want `-i` for keys

File: continuous-upgrade/jobs/cicd-upgrade-ocp.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+rpm -qa atomic-openshift
+sudo python sjb/hack/determine_install_upgrade_version.py "\$( rpm -qa atomic-openshift )" > AOS_VARS
+source AOS_VARS
+ansible-playbook  -vv                    \
+                  --become               \
+                  --become-user root     \
+                  --connection local     \
+                  --inventory sjb/inventory/ \
+                   "/usr/share/ansible/openshift-ansible/playbooks/byo/openshift-cluster/upgrades/v3_${ATOMIC_OPENSHIFT_UPGRADE_RELEASE_MINOR_VERSION=6}/upgrade.yml"     \
+                  -e etcd_data_dir="/tmp/etcd" \
```
Comment (stevekuznetsov) [inline continuous-upgrade/jobs/cicd-upgrade-ocp.sh]: Don't think we need this

File: continuous-upgrade/jobs/cicd-upgrade-ocp.sh [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+sudo yum --disablerepo=* --enablerepo=openshift-int,oso-rhui-rhel-server-releases update -y atomic-openshift-utils
+cd /root/aos-cd-jobs/
+rpm -qa atomic-openshift
+sudo python sjb/hack/determine_install_upgrade_version.py "\$( rpm -qa atomic-openshift )" > AOS_VARS
+source AOS_VARS
+ansible-playbook  -vv                    \
+                  --become               \
+                  --become-user root     \
+                  --connection local     \
+                  --inventory sjb/inventory/ \
```
Comment (stevekuznetsov) [inline continuous-upgrade/jobs/cicd-upgrade-ocp.sh]: Isn't there a special inventory on this host somewhere for us to use?

**Comment by jhadvig** [comment]
@stevekuznetsov PTAL

**Comment by stevekuznetsov** [comment]
Seems reasonable -- is the continuous job getting artifacts as necessary now?

**Comment by jhadvig** [comment]
@stevekuznetsov I've updated the PR. The jUnit plugin is installed for the cicd cluster is installed upon the setup process in [this PR](https://github.com/openshift/openshift-ansible-ops/pull/2789). In the single node cluster case its installed during the provision job. 
The cicd upgrade process itself is running smoothly the last thing I need to figure out is why we cant access the junit logs even thought Im copying them to the jobs `$WORKSPACE`  as you can see the the bottom of [this job](https://ci.openshift.redhat.com/jenkins/view/All/job/cicd_upgrade-job/37/console)

PTAL

**Comment by stevekuznetsov** [comment]
Not sure what you mean -- I see [jUnit logs](https://ci.openshift.redhat.com/jenkins/view/All/job/cicd_upgrade-job/37/testReport/) as expected for the build

**Comment by jhadvig** [comment]
oh... sorry man... Im getting blind :smile:

**Comment by jhadvig** [comment]
@stevekuznetsov we need to merge this since the [single node cluster continuous upgrade job](https://ci.openshift.redhat.com/jenkins/view/All/job/continuous-upgrade_provision-job/21/console) is failing casue after tcloning the repo ist missing the install-junit script

**Pull request merged**